mavros/src/plugins/wind_estimation.cpp

```diff
@@ -68,21 +68,26 @@ class WindEstimationPlugin : public plugin::PluginBase {
 		twist_cov->twist.twist.linear.y = speed * std::cos(course);
 		twist_cov->twist.twist.linear.z = wind.speed_z;
 
+		// fill available covariance elements (-1 = unknown)
+		for (int i = 0; i < 36; ++i) twist_cov->twist.covariance[i] = -1.0;
```

Comment:
Use `std::array` and `std::copy()` instead. And only index 0 is to be set to -1. The others should be 0.

Reply:
You want to represent a *complete* unknown covariance matrix by only putting -1 in the first element of the array (and the rest zeros), and then on the other (PX4) implementation, just fill unknowns with zeros? I dont understand, can you clarify please? I thought you wanted any unknown as a -1 (which is all for APM, and all but two for PX4).

Reply:
ROS convention states that only the first value of the array (covariance matrix) should be valued -1 when the values of the covariance matrix are unknown or invalid. There is no convention for the autopilot side or to MAVLink, though I am also planning to open a RFC so that can be discussed.

Reply:
Where is that convention for unkown covariance matrix values stated on the autopilot side? I am not aware of those. For MAVLink that does not exist.

Reply:
I have no idea. That's why I asked. I originally just assumed one would default the values to zero if not populated. Was just confused at the first comment. I'll implement with the ROS convention for APM message (since there is no covariance getting sent in the msg), and fill the numbers present in the PX4 message, and the rest zeros. Let me know if I'm still not getting it right.

Reply:
Well probably that need some clarification. Can you open an issue on PX4 so someone can clarify it? I suppose the answer should be in the wind estimator for that case, but better having someone clarify it. But yes for APM makes sense to apply the ROS convention. For PX4, the question remains regarding the XY components.